### ISSUE TITLE
RequireJs documentation is too wordy 

### ISSUE THREAD

**Issue by TheSpiderPig** [jrburke/requirejs#1261]
**I LOVE RequireJs** Browserfy is going to kick RequireJs's ass because its easier to learn even though (IMHO) requirejs is better. The reason: RequireJs's documentation is **too wordy and bloated and doesn't focus enough on code**  

It goes too far into depth in explaining concepts and not enough in explaining the BARE BONES of what is the API in the first place.
Additionally the font style and usage of *italics* makes it really difficult to skim for the important aspects of what you need to know. 

Essentially the format should be reformatted into:
1. Function definition
2. **BRIEF** summary of what the function does 
3. Explain parameters 
4. Example of how to use it 
5. Additional explanation or discussion topics such as best practices or how it interacts with other APIs. (Marionett JS would be a good example of this part)

### define(dependencies, moduleDefinitionFunction)

defines a new module. 

**dependencies** - Array of module names or file paths to other modules 
**moduleDefinitionFunction** - will returns an object that will be imported into other modules. The function parameters map one to one to the dependencies required.
  
     //my_module.js
     define([ 'jquery]', function($) {
         // module initialization code 
         var x = $('.someSelector');
      
        function MyObjectDefinition() {
             this.name = 5;
         }
         MyObjectDefinition.prototype.someFunction = function() {
             ///do something   
         }
        return {
            x: selector, 
            MyObjectDefinition: MyObjectDefinition
        };
      });
    
You can require this module into other modules by file name whose root is defined by your require.config 

    define(['./my_module'], function(MyModule) {
        var instance = new MyModule.MyObjectDefinition();
    }

**More explanation of whats going on here, best practices, common practices, use cases and/or edge cases **

**Comment by jrburke** [comment]
This would be helpful. There is #567 that was previously filed, getting to the same idea, so will mark this as a duplicate (but this bug will be referenced in the other bug, so part of the info for that bug).
